openqabot/types/aggregate.py

```diff
@@ -57,157 +57,159 @@ def get_buildnr(repohash: str, old_repohash: str, build: str) -> str:
+            ).json()
+        except Exception as e:
+            log.exception(e)
+            old_jobs = None
+
+        old_repohash = old_jobs[0].get("repohash", "") if old_jobs else ""
+        old_build = old_jobs[0].get("build", "") if old_jobs else ""
 
-            # add to ret
-            ret.append(full_post)
+        try:
+            full_post["openqa"]["BUILD"] = self.get_buildnr(
+                full_post["openqa"]["REPOHASH"], old_repohash, old_build
+            )
+        except SameBuildExists:
+            log.info(
+                "For %s aggreagate on %s there is existing build" % (self.product, arch)
+            )
+            return {}
+
+        if not ignore_onetime and (
+            self.onetime and full_post["openqa"]["BUILD"].split("-")[-1] != "1"
+        ):
+            return {}
+
+        settings = self.settings.copy()
+
+        # if set, we use this query to detect latest public cloud tools image which used for running
+        # all public cloud related tests in openQA
+        if "PUBLIC_CLOUD_TOOLS_IMAGE_QUERY" in settings:
+            query = settings["PUBLIC_CLOUD_TOOLS_IMAGE_QUERY"]
+            settings = apply_pc_tools_image(settings)
+            if not settings.get("PUBLIC_CLOUD_TOOLS_IMAGE_BASE", False):
+                log.error(
+                    f"Failed to query latest publiccloud tools image using {query}"
+                )
+                return {}
+
+        # parse Public-Cloud image REGEX if present
+        if "PUBLIC_CLOUD_IMAGE_REGEX" in settings:
+            settings = apply_publiccloud_regex(settings)
+            if not settings.get("PUBLIC_CLOUD_IMAGE_LOCATION", False):
+                log.error(
+                    f"No publiccloud image found for {settings['PUBLIC_CLOUD_IMAGE_REGEX']}"
+                )
+                return {}
+        # parse Public-Cloud pint query if present
+        if "PUBLIC_CLOUD_PINT_QUERY" in settings:
+            settings = apply_publiccloud_pint_image(settings)
+            if not settings.get("PUBLIC_CLOUD_IMAGE_ID", False):
+                log.error(
+                    f"No publiccloud image fetched from pint for for {settings['PUBLIC_CLOUD_PINT_QUERY']}"
+                )
+                return {}
+
+        full_post["openqa"].update(settings)
+        full_post["openqa"]["FLAVOR"] = self.flavor
+        full_post["openqa"]["ARCH"] = arch
+        full_post["openqa"]["_OBSOLETE"] = 1
+
+        for template, issues in test_incidents.items():
+            full_post["openqa"][template] = ",".join(str(x) for x in issues)
+        for template, issues in test_repos.items():
+            full_post["openqa"][template] = ",".join(issues)
+        for issues in test_incidents.values():
+            full_post["qem"]["incidents"] += issues
+
+        full_post["qem"]["incidents"] = [
+            str(inc) for inc in set(full_post["qem"]["incidents"])
+        ]
+
+        if not full_post["qem"]["incidents"]:
+            return {}
```

Comment:
This will ultimately create a different list than in the original code. Instead of skipping invalid items, this will add an empty one, e.g. you can end up with something like: `[{},{},{}]`